app/liquid/liquid_renderer.rb

```diff
@@ -43,9 +51,48 @@ def images
 
   private
 
+  def outstanding_fields(plugin_data)
+    isolate_from_plugin_data(plugin_data, :outstanding_fields)
+  end
+
+  def donation_bands(plugin_data)
+    isolate_from_plugin_data(plugin_data, :donation_bands).first
+  end
+
+  def isolate_from_plugin_data(plugin_data, field)
+    plugin_values = plugin_data.deep_symbolize_keys[:plugins].values().map(&:values).flatten
+    plugin_values.map{|plugin| plugin[field]}.flatten.compact
+  end
+
+  def location
+    return @location if @location.blank?
+    country_code = if @member.try(:country) && @member.country.length == 2
+      @member.country
+    else
+      @location.country_code
+    end
+    return @location.data if country_code.blank?
+    currency = Donations::Utils.currency_from_country_code(country_code)
+    @location.data.merge(currency: currency)
+  end
+
+  def set_locale
+    begin
+      I18n.locale = @page.language.code if @page.language.present?
+    rescue I18n::InvalidLocale
+      # by setting the +i18n.enforce_available_locales+ flag to true but
+      # catching the resulting error, it allows us to only set the locale
+      # if it's one explicitly registered under +i18n.available_locales+
+    end
+  end
+
   def image_urls(img)
     return { urls: { large: '', small: '' } } if img.blank? || img.content.blank?
     { urls: { large: img.content.url(:large), small: img.content.url(:thumb) } }
   end
 
+  def cache_key
+    "rendered_liquid:#{@page.cache_key}:#{@page.liquid_layout.cache_key}"
```

Comment:
this doesn't work as intended because the cache key is the same whether you're rendering the liquid layout or the secondary liquid layout. I'm changing `@page.liquid_layout.cache_key` to `@liquid_layout.cache_key`.